src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -640,7 +640,10 @@ def load_arguments(self, _):
     # region VM & VMSS Shared
     for scope in ['vm', 'vmss']:
         with self.argument_context(scope) as c:
-            c.argument('no_auto_upgrade', arg_type=get_three_state_flag(), help='If set, the extension service will not automatically pick or upgrade to the latest minor version, even if the extension is redeployed.')
+            c.argument('no_auto_upgrade',
+                       options_list=['--no-auto-upgrade-mi', c.deprecate(target='--no-auto-upgrade', redirect='--no-auto-upgrade-mi')],
```

Comment:
for options_list, I would suggest to keep sync with Swagger property name so user has consistent context. Since Swagger properties `autoUpgradeMinorVersion` and `enableAutomaticUpgrade` is confusing enough already. I can see the difficult here. `--no-auto-upgrade` is `!autoUpgradeMinorVersion`.  How about `--no-auto-upgrade-minorversion`?

Reply:
hi @qwordy , what's future plan of original param `no-auto-upgrade`, If it keeps there, this will not be a breaking change.

Reply:
Yes. It is really confusing. I suggest keep "no" prefix for it is already here. "--no-auto-upgrade-minorversion" is too long. It can't pass linter. We need a better name to express the meaning clearly and not be very long. How about "--no-aum"?
Yes. The old parameter can still be used. It is not a breaking change for now.

Reply:
Agree that `--no-auto-upgrade-minorversion` is long. While first thing here is clarity. I have one question: If user set `--no-auto-upgrade-mi` to true,  while `--enable-auto-upgrade` true. What will happen when there's new minor version released of extension? WIll `enableAutomaticUpgrade` overwrite `autoUpgradeMinorVersion` value?